### PR TITLE
New: Add parser for TypeScript and TSX files

### LINKED ISSUE #2888: Support TypeScript files

### PR DESCRIPTION
**Pull request by antross**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [x] Added/Updated related documentation.
- [x] Added/Updated related tests.

## Short description of the change(s)

* Allow existing JS hints to run against *.ts and *.tsx files
* Allow existing HTML hints to run against TSX content

- - - - - - - - - -

Fix #2888

### CODE REVIEW

File: packages/parser-typescript/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# TypeScript (`@hint/parser-typescript`)
+
+The `typescript` parser emits the same events as `@hint/parser-javascript`
+so existing hints targeting JavaScript files can analyze TypeScript files
+without modification. Existing hints targeting HTML files can also analyze
+TSX content without modification if `@hint/parser-jsx` is active in addition
+to this parser.
```
Comment (molant) [inline packages/parser-typescript/README.md]: 😍

File: packages/parser-typescript/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    "exclude": [
+        "dist",
+        "node_modules"
+    ],
+    "extends": "../../tsconfig.json",
+    "include": [
+        "src/**/*.ts",
+        "tests/**/*.ts"
+    ],
+    "references": [
```
Comment (molant) [inline packages/parser-typescript/tsconfig.json]: Do we need to update the global references? We tend to forget them but I'm not sure when they are needed.
Maybe we should add a step in `create-hint/parser` packages that if it's in the official package it automatically runs the update dependencies script?

Comment (antross) [reply packages/parser-typescript/tsconfig.json]: Yep, forgot to do that. Just pushed a commit with the update - I'll rebase and splice it in once we're happy with the rest of the PR.

To make `update:dependencies` work correctly we need to re-run it each time our dependencies in `package.json` change too. Maybe we should just make this part of `build` or `test` for all our packages.

Comment (molant) [reply packages/parser-typescript/tsconfig.json]: I'm happy with the rest of the PR and @sarvaje as well. If you want to go ahead and merge feel free!

Comment (antross) [reply packages/parser-typescript/tsconfig.json]: Ok, I'm going to rebase on `master` and splice in that last commit. Once CI passes after that update I'll go ahead and merge.

**Comment by molant** [comment]
For future reference, here's an example of the VS Code extension working with TSX files.

webhint analyzing a TSX file on VS Code

**Pull request merged**